File: src/com/rian/difficultycalculator/beatmap/hitobject/SliderPathType.java
package com.rian.difficultycalculator.beatmap.hitobject;

/**
 * Types of slider paths.
 */
public enum SliderPathType {
    Catmull,
    Bezier,
    Linear,
    PerfectCurve;

    public static SliderPathType parse(char value) {
        return switch (value) {
            case 'C' -> Catmull;
            case 'L' -> Linear;
            case 'P' -> PerfectCurve;
            default -> Bezier;
        };
    }
}

File: src/com/rian/osu/beatmap/hitobject/SliderPathType.kt
package com.rian.osu.beatmap.hitobject

/**
 * Types of slider paths.
 */
enum class SliderPathType {
    Catmull,
    Bezier,
    Linear,
    PerfectCurve;

    companion object {
        @JvmStatic
        fun parse(value: Char) =
            when (value) {
                'C' -> Catmull
                'L' -> Linear
                'P' -> PerfectCurve
                else -> Bezier
            }
    }
}
